win32ss/user/ntuser/caret.c

```diff
@@ -300,7 +288,16 @@ BOOL FASTCALL co_UserShowCaret(PWND Window OPTIONAL)
       {
          IntNotifyWinEvent(EVENT_OBJECT_SHOW, pWnd, OBJID_CARET, OBJID_CARET, 0);
       }
-      IntSetTimer(pWnd, IDCARETTIMER, gpsi->dtCaretBlink, CaretSystemTimerProc, TMRF_SYSTEM);
+      if ((INT)gpsi->dtCaretBlink > 0)
+      {
+          IntSetTimer(pWnd, IDCARETTIMER, gpsi->dtCaretBlink, CaretSystemTimerProc, TMRF_SYSTEM);
+      }
+      else if (ThreadQueue->CaretInfo.Visible)
+      {
+          ThreadQueue->CaretInfo.Showing = 1;
+          co_IntDrawCaret(pWnd, &ThreadQueue->CaretInfo);
+      }
+      IntNotifyWinEvent(EVENT_OBJECT_SHOW, pWnd, OBJID_CARET, OBJID_CARET, 0);
```

Comment:
Should this event be broadcast if you are starting the timer, or just when the thing is made visible?

Reply:
Of cource. Thanks for your review.